optaplanner-core/src/test/java/org/optaplanner/core/impl/testdata/domain/shadow/cyclic/TestdataSevenNonCyclicShadowedEntity.java

```diff
@@ -73,7 +73,7 @@ public void setValue(TestdataValue value) {
         this.value = value;
     }
 
-    @CustomShadowVariable(variableListenerClass = VariableListenerAdapter.class, sources = {
+    @CustomShadowVariable(variableListenerClass = VariableListener.class, sources = {
```

Comment:
It's wierd that this doesn't trigger a fail-fast because it's not a concrete class it can instantiate.
For this reason, we might want to keep VariableListenerAdapter in src/test/java as DummyVariableListener

Reply:
`VariableListenerAdapter` used to be `abstract`, so there's no direct instances of that either. Status quo does not change.